language/move-prover/bytecode/src/compositional_analysis.rs

```diff
@@ -57,20 +57,21 @@ impl<'a> SummaryCache<'a> {
 }
 
 /// Trait that lifts an intraprocedural analysis into a bottom-up, compositional interprocedural
-/// analysis. The derived `summarize` function computes a postcondition for the procedure under
-/// analysis, then stores it in the summary cache for other procedures to use.
-// TODO: allow client analyses to customize this by providing a function that converts a
-// postcondition into a summary
+/// analysis.
 pub trait CompositionalAnalysis: DataflowAnalysis
 where
-    Self::State: 'static,
+    Self::State: AbstractDomain + 'static,
 {
-    fn summarize(
+    /// `analysis.summarize_abstract env init data abstrct` computes a postcondition for the
```

Comment:
Doc comments will show the parameter names, so I think you can start the comment at "Computes a postcondition"

Reply:
Also, if you actually want to mention the call in the documentation comment, please use correct syntax. You example would be `self.summarize_abstract(env, init, data, abstrct)`.